### ISSUE TITLE
Consider switching defined classes/methods from public to open

### ISSUE THREAD

**Issue by fim** [IBM-Swift/Kitura#937]
Traditionally, swift allowed both using and subclassing/overriding classes and methods that are defined as public. However, since version 3.0 this is no longer the case and public only allows using the public classes/methods whereas open allows both.

With the current implementation, something like the following:

```swift
class MyStaticFileServer: StaticFileServer {
  override func handle(request: RouterRequest, response: RouterResponse, next: @escaping () -> Void {
    // do something
    super.handle(request: request, response: response, next: next)
  }
}
```

throws the following error:

```sh
test.swift:38:18: error: overriding non-open instance method outside of its defining module
            override func handle(request: RouterRequest, response: RouterResponse, next: @escaping () -> Void) {

```

This is when running with:

```sh
Apple Swift version 3.0.2 (swiftlang-800.0.63 clang-800.0.42.1)
Target: x86_64-apple-macosx10.9
```

The following changes make the error go away:

```diff
diff --git a/Sources/Kitura/staticFileServer/StaticFileServer.swift b/Sources/Kitura/staticFileServer/StaticFileServer.swift
index ee47a99..9ce9951 100644
--- a/Sources/Kitura/staticFileServer/StaticFileServer.swift
+++ b/Sources/Kitura/staticFileServer/StaticFileServer.swift
@@ -19,7 +19,7 @@ import Foundation
 // MARK: StaticFileServer

 /// A router middleware that serves static files from a given path.
-public class StaticFileServer: RouterMiddleware {
+open class StaticFileServer: RouterMiddleware {

     /// Cache configuration options for StaticFileServer.
     public struct CacheOptions {
@@ -101,7 +101,7 @@ public class StaticFileServer: RouterMiddleware {
     /// - Parameter request: the router request.
     /// - Parameter response: the router response.
     /// - Parameter next: the closure for the next execution block.
-    public func handle(request: RouterRequest, response: RouterResponse, next: @escaping () -> Void) {
+    open func handle(request: RouterRequest, response: RouterResponse, next: @escaping () -> Void) {
         guard request.serverRequest.method == "GET" || request.serverRequest.method == "HEAD" else {
             return next()
         }
```

Please consider moving classes and methods to open rather than public unless this limitation is by design. More information about this change can be found here: https://github.com/apple/swift-evolution/blob/master/proposals/0117-non-public-subclassable-by-default.md

Thanks!

**Comment by shmuelk** [comment]
This sounds like a good idea. However, we need to get Jazzy, the API documentation HTML generator fixed first, as apparently it doesn't add functions declared as open to the outputed HTML.

**Comment by ianpartridge** [comment]
Jazzy has now been fixed so I think this is worth pursuing, for the usecase of `StaticFileServer`.  Assigning to @Andrew-Lees11 to tweak these access modifiers.

**Comment by ianpartridge** [comment]
Fixed in #1245
